core/src/main/scala/spire/optional/Perm.scala

```diff
@@ -0,0 +1,62 @@
+package spire
+package optional
+
+import scala.collection.compat._
+import spire.scalacompat.SeqLike
+
+import cats.kernel.Eq
+import spire.algebra.{Action, Group}
+import spire.algebra.partial.PartialAction
+import spire.syntax.cfor._
+import spire.syntax.group._
+import spire.util._
+
+/**
+ * Represents a permutation encoded as a map from preimages to images, including
+ * only pairs that are moved by the permutation (so the identity is Map.empty).
+ */
+case class Perm(mapping: Map[Int, Int]) extends AnyVal {
+  def apply(k: Int): Int = mapping.getOrElse(k, k)
+  def image: Set[Int] = mapping.keySet
+  def invert(k: Int) = mapping.find(_._2 == k).fold(k)(_._1)
+}
+
+object Perm {
+  implicit val PermEq: Eq[Perm] = Eq.fromUniversalEquals
```

Comment:
This is new. Formerly the test that needed an `Eq[Map[Int, Int]]` was getting it from [`vectorOrder`](https://github.com/typelevel/spire/blob/3446d41fbbf3c0a117b4b5c86fe0ec2e7d539fd5/core/src/main/scala/spire/optional/vectorOrder.scala#L17)

Reply:
Technically, a Map(1 -> 1, 2 -> 2, ..., n -> n) should be the same as an empty Map (they're all the identity), but the group combine method eliminates those redundant cases. I could make it a requirement of the Perm constructor, rather than normalizing here.